boards/waspmote-pro/board.c

```diff
@@ -58,10 +58,6 @@ void board_init(void)
 #endif
 #endif
 
-#ifdef MODULE_AVR_LIBC_EXTRA
-    atmega_stdio_init();
-#endif
-
     cpu_init();
```

Comment:
On some architectures it was initialized after `periph_init()`. I added a bunch of PRs to consistently initialize it just before `periph_init()`. As `periph_init()` is hardware independent (as it is in `drivers`), we can only rely on using `DEBUG()` there if independent of hardware `stdio` is available there.

So there were to goals:
1. Move the point where `stdio` is guaranteed to be working to a point before `periph_init()` (Does not apply to AVR, was already the case)
2. Have `stdio` available at a consistent point during boot up to avoid confusion.

For the ATmegas only point 2 applied, but it also resulted in less code duplication :-)